piptools/resolver.py

```diff
@@ -75,7 +75,17 @@ def resolve_hashes(self, ireqs):
         """
         Finds acceptable hashes for all of the given InstallRequirements.
         """
-        return {ireq: self.repository.get_hashes(ireq) for ireq in ireqs}
+        hashes = {}
+        for ireq in ireqs:
+            try:
+                ireq_hash = self.repository.get_hashes(ireq)
+            except TypeError:
+                continue
+            else:
+                if ireq_hash is not None:
```

Comment:
I could not find a case where the result would be `None`. Di you find one?
Also, I'm not sure I'd like to silently skip hashes, except for clear, defined cases, so I would rather see it fail here. Otherwise, that might bring up a false sense of security, which is really not nice.